### PR TITLE
Enrico/fix parsing exceptions

### PR DESCRIPTION
**Pull request by EnricoTrentin**
In this change:
- Fix all exception encountered while parsing legend-pure-m3-core (a lot!)
- Fix startup script in main.js for mac os.

### CODE REVIEW

**Comment by db-ci-telescope** [comment]
<ID><b><u>Diffblue Cover:</u></b> enrico/fixParsingExceptions<br><TEST-GEN><b>Test Generation Status:</b> Diffblue is analysing your pull request :hourglass_flowing_sand:<br>

File: core/src/main/java/com/diffblue/telescope/reports/jacocoReport/JacocoReportParser.java [hunk 1]
```diff
@@ -178,22 +180,107 @@ private ClassMetrics getClassMetrics(Node currentClass, SourceFile sourceFile) {
...
                 classMetrics.addMetric(currentAtt.getAttributes().getNamedItem("type").getTextContent() + "-missed", Integer.parseInt(currentAtt.getAttributes().getNamedItem("missed").getTextContent()));
                 classMetrics.addMetric(currentAtt.getAttributes().getNamedItem("type").getTextContent() + "-covered", Integer.parseInt(currentAtt.getAttributes().getNamedItem("covered").getTextContent()));
             }
         }
+
         return classMetrics;
     }
 
+    /**
+     * @param currentClass Xml node containing all data about this class
```
Comment (EnricoTrentin) [inline core/src/main/java/com/diffblue/telescope/reports/jacocoReport/JacocoReportParser.java:195]: update the javadocs

File: core/src/main/java/com/diffblue/telescope/reports/jacocoReport/JacocoReportParser.java [hunk 2]
```diff
@@ -178,22 +180,107 @@ private ClassMetrics getClassMetrics(Node currentClass, SourceFile sourceFile) {
...
+
+        List<ClassMetrics> classesMetrics = subClasses.stream()
+            .map(subclass -> getClassMetrics(subclass, sourceFile))
+            .sorted((cl1, cl2) -> {
+                int classFirstLine1 = getFirstLine(cl1);
+                int classFirstLine2 = getFirstLine(cl2);
+                if (classFirstLine1 != classFirstLine2) {
+                    return classFirstLine1<classFirstLine2?1:-1; // reversed firstLine order
+                } else {
+                    try {
```
Comment (EnricoTrentin) [inline core/src/main/java/com/diffblue/telescope/reports/jacocoReport/JacocoReportParser.java:210]: Number of lines does not discriminate inner and outer classes

File: core/src/main/java/com/diffblue/telescope/reports/jacocoReport/JacocoReportParser.java [hunk 3]
```diff
@@ -178,22 +180,107 @@ private ClassMetrics getClassMetrics(Node currentClass, SourceFile sourceFile) {
...
     }
 
+    /**
+     * @param currentClass Xml node containing all data about this class
+     * @return A ClassMetrics object containing all data about the current class and methods it contains
+     */
+    private List<ClassMetrics> getClassesMetrics(Node currentClass, List<Node> subClasses, SourceFile sourceFile) {
+
+        ClassMetrics parentClassMetrics = getClassMetrics(currentClass, sourceFile);
+
```
Comment (EnricoTrentin) [inline core/src/main/java/com/diffblue/telescope/reports/jacocoReport/JacocoReportParser.java:201]: explain why subclasses must be sorted

File: core/src/main/java/com/diffblue/telescope/reports/jacocoReport/JacocoReportParser.java [hunk 4]
```diff
@@ -207,49 +294,125 @@ private MethodMetrics getMethodMetrics(String className, Node currentMethod, Sou
...
+                    Map<String, Integer> methodTracker = new HashMap<>();
+                    methodTracker.put("firstLine", (Integer)method.getMetric("firstLine"));
+                    methodTracker.put("lines",     (Integer)method.getMetric("LINE-missed") 
+                                                 + (Integer)method.getMetric("LINE-covered"));
+                    methodTracker.put("missed",    (Integer)method.getMetric("INSTRUCTION-missed"));
+                    methodTracker.put("covered",   (Integer)method.getMetric("INSTRUCTION-covered"));
+                    return methodTracker;
+                }
+            ));
+
```
Comment (EnricoTrentin) [inline core/src/main/java/com/diffblue/telescope/reports/jacocoReport/JacocoReportParser.java:336]: compare number of instructions in linestracker and methodstracker

File: core/src/main/java/com/diffblue/telescope/reports/jacocoReport/JacocoReportParser.java [hunk 5]
```diff
@@ -207,49 +294,125 @@ private MethodMetrics getMethodMetrics(String className, Node currentMethod, Sou
...
-
-        // Use the starting line number from the method to create an offset so the corresponding lines are taken from the source file
-        int firstLine = Integer.parseInt(currentMethod.getAttributes().getNamedItem("line").getTextContent());
-        int offset = 0;
-        boolean isFirstLine = true;
-
-        int methodLines = (int) methodMetrics.getMetric("LINE-missed") + (int) methodMetrics.getMetric("LINE-covered");
-        int lineCount = lineCoverage.size();
-
-        if (methodLines > lineCount) {
```
Comment (EnricoTrentin) [inline core/src/main/java/com/diffblue/telescope/reports/jacocoReport/JacocoReportParser.java:232]: Add a check to make sure linetracker has been completely used

**Review by ThomasPerkins1123 — APPROVED** [review]

**Pull request merged**